grpc-server/workflow.go

```diff
@@ -267,35 +268,29 @@ func createYaml(ctx context.Context, sqlDB *sql.DB, temp string, tar string) (st
 	if err != nil {
 		return "", err
 	}
-	tarData, err := db.TargetsByID(ctx, sqlDB, tar)
-	if err != nil {
-		return "", err
-	}
-	return renderTemplate(string(tempData), []byte(tarData))
+	return renderTemplate(string(tempData), []byte(tar))
 }
 
 func renderTemplate(tempData string, tarData []byte) (string, error) {
-	type machine map[string]string
-	type Environment struct {
-		Targets map[string]machine `json:"targets"`
-	}
-
-	var env Environment
-	t := template.New("workflow-template")
-	_, err := t.Parse(tempData)
+	var hardware map[string]interface{}
```

Comment:
Is there any specific reason why we are using `map[string]interface{}` instead of `map[string]string`?
Because in this case the following will also work, while it is incorrect:
```
$ tink workflow create -t <template-id> -r '{"device_1": 12345}'
```

Reply:
I understand this will cover cases like:
```
{"device_one": {"mac": "macaddr"}, "device_two": "device-two-value"}
```

However, we must add a few validations around what is being provided as hardware value at the time of workflow creation.

Reply:
`interface{}` is being used as a value because in future there is a possibility there can be nested dirs in the input and you have to fetch the mac addr or IP addr from the nested key:value pair. In that case you will just need to change the template and you will be able to create the workflow without changing the code. But yes some validation must be required before placing them in the workflow.